internal/gatewayapi/testdata/httproute-with-mirror-filter-percentage-mirroring.in.yaml

```diff
@@ -0,0 +1,53 @@
+gateways:
+- apiVersion: gateway.networking.k8s.io/v1
+  kind: Gateway
+  metadata:
+    namespace: envoy-gateway
+    name: gateway-1
+  spec:
+    gatewayClassName: envoy-gateway-class
+    listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+      hostname: "*.envoyproxy.io"
+      allowedRoutes:
+        namespaces:
+          from: All
+httpRoutes:
+- apiVersion: gateway.networking.k8s.io/v1
+  kind: HTTPRoute
+  metadata:
+    namespace: default
+    name: httproute-1
+  spec:
+    hostnames:
+    - gateway.envoyproxy.io
+    parentRefs:
+    - namespace: envoy-gateway
+      name: gateway-1
+      sectionName: http
+    rules:
+    - matches:
+      - path:
+          value: "/"
+      backendRefs:
+      - name: service-1
+        port: 8080
+      filters:
+      - type: RequestMirror
+        requestMirror:
+          backendRef:
+            kind: Service
+            name: service-1
+            port: 8080
+          percent: 20
+      - type: RequestMirror
+        requestMirror:
+          backendRef:
+            kind: Service
+            name: mirror-service
+            port: 8080
+          fraction:
+            numerator: 25
+            denominator: 50
```

Comment:
nit: can we change the denominator to make this percent less than 1% ?

Reply:
We should be. in this case 5000 would make that.  any complications in envoy that I am not aware of with less than 1%?

Reply:
Not that I'm aware of, just making sure the code can deal with less than 0 here and we'll need a conformance/e2e to make sure data plane can deal with it as well

Reply:
Do you want more tests here? I can add tests in gateway upstream for less-than-one percent.

Reply:
Nah this is fine, hoping the upstream conformance tests for less than 1 percent